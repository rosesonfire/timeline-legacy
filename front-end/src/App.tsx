import React, { useEffect, useState } from 'react';
import {
  Text,
  Link,
  HStack,
  Center,
  ScrollView,
  Heading,
  Switch,
  useColorMode,
  NativeBaseProvider,
  extendTheme,
  VStack,
  Box,
} from 'native-base';

import { createEntity, getEntities } from 'api/entity';

import NativeBaseIcon from 'components/NativeBaseIcon';

// Define the config
const config = {
  useSystemColorMode: false,
  initialColorMode: 'dark',
};

// extend the theme
export const theme = extendTheme({ config });
type MyThemeType = typeof theme;
declare module 'native-base' {
  interface ICustomTheme extends MyThemeType {}
}

export default function App() {
  const [response, setResponse] = useState('');

  useEffect(() => {
    createEntity({
      name: 'some name',
    });

    getEntities()
      .then((res) => {
        setResponse(JSON.stringify(res.data));
      })
      .catch((error) => {
        setResponse(JSON.stringify(error, null, 2));
      });
  }, []);

  return (
    <NativeBaseProvider>
      <ScrollView>
        <Center _dark={{ bg: 'blueGray.900' }} _light={{ bg: 'blueGray.50' }} px={4} flex={1}>
          <VStack space={5} alignItems="center">
            <NativeBaseIcon />
            <Heading size="lg">Welcome to NativeBase</Heading>
            <Heading size="md">{response}</Heading>
            <HStack space={2} alignItems="center">
              <Text>Edit</Text>
              <Box
                _web={{
                  _text: {
                    fontFamily: 'monospace',
                    fontSize: 'sm',
                  },
                }}
                px={2}
                py={1}
                _dark={{ bg: 'blueGray.800' }}
                _light={{ bg: 'blueGray.200' }}
              >
                App.js
              </Box>
              <Text>and save to reload.</Text>
            </HStack>
            <Link href="https://docs.nativebase.io" isExternal>
              <Text color="primary.500" underline fontSize={'xl'}>
                Learn NativeBase
              </Text>
            </Link>
            <ToggleDarkMode />
          </VStack>
        </Center>
      </ScrollView>
    </NativeBaseProvider>
  );
}

// Color Switch Component
function ToggleDarkMode() {
  const { colorMode, toggleColorMode } = useColorMode();
  return (
    <HStack space={2} alignItems="center">
      <Text>Dark</Text>
      <Switch
        isChecked={colorMode === 'light'}
        onToggle={toggleColorMode}
        aria-label={colorMode === 'light' ? 'switch to dark mode' : 'switch to light mode'}
      />
      <Text>Light</Text>
    </HStack>
  );
}
