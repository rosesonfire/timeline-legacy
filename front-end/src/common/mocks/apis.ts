jest.mock('api/entity', () => ({
  createEntity: jest.fn(),
  getEntities: () => Promise.resolve({ data: 'data' }),
}));
