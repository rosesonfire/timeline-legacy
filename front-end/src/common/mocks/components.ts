jest.mock('components/NativeBaseIcon', () => 'NativeBaseIcon');
