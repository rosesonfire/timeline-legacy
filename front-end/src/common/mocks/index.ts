import './vendors';
import './components';
import './apis';
