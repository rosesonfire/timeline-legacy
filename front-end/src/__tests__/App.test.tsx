import React from 'react';
import renderer from 'react-test-renderer';

import '@common/mocks';

import App from '../App';

describe('App', () => {
  it('should match snapshot', () => {
    const actual = renderer.create(<App />);

    expect(actual).toMatchSnapshot();
  });
});
