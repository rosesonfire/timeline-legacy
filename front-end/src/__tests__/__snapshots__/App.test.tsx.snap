// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<NativeBaseProvider3>
  <ScrollView>
    <Center
      _dark={
        {
          "bg": "blueGray.900",
        }
      }
      _light={
        {
          "bg": "blueGray.50",
        }
      }
      flex={1}
      px={4}
    >
      <VStack
        alignItems="center"
        space={5}
      >
        <NativeBaseIcon />
        <Heading
          size="lg"
        >
          Welcome to NativeBase
        </Heading>
        <Heading
          size="md"
        />
        <HStack
          alignItems="center"
          space={2}
        >
          <Text>
            Edit
          </Text>
          <Box
            _dark={
              {
                "bg": "blueGray.800",
              }
            }
            _light={
              {
                "bg": "blueGray.200",
              }
            }
            _web={
              {
                "_text": {
                  "fontFamily": "monospace",
                  "fontSize": "sm",
                },
              }
            }
            px={2}
            py={1}
          >
            App.js
          </Box>
          <Text>
            and save to reload.
          </Text>
        </HStack>
        <Link
          href="https://docs.nativebase.io"
          isExternal={true}
        >
          <Text
            color="primary.500"
            fontSize="xl"
            underline={true}
          >
            Learn NativeBase
          </Text>
        </Link>
        <HStack
          alignItems="center"
          space={2}
        >
          <Text>
            Dark
          </Text>
          <Switch
            aria-label="switch to dark mode"
            isChecked={true}
            onToggle={[MockFunction]}
          />
          <Text>
            Light
          </Text>
        </HStack>
      </VStack>
    </Center>
  </ScrollView>
</NativeBaseProvider3>
`;
